sequencer/src/external_event_handler.rs

```
@@ -105,73 +78,63 @@ impl<V: Versions> ExternalEventHandler<V> {
+                    .await?;
             }
         }
         Ok(())
     }
 
-    /// Creates a roll call response message
-    async fn create_roll_call_response(
-        public_key: &BLSPubKey,
-        roll_call_info: &RollCallInfo,
-    ) -> Result<Vec<u8>> {
-        let response = ExternalMessage::RollCallResponse(roll_call_info.clone());
-
-        // Serialize the response
-        let response_bytes = bincode::serialize(&response)
-            .with_context(|| "Failed to serialize roll call response")?;
-
-        let message = Message::<SeqTypes> {
-            sender: *public_key,
-            kind: MessageKind::<SeqTypes>::External(response_bytes),
-        };
-
-        let response_bytes = bincode::serialize(&message)
-            .with_context(|| "Failed to serialize roll call response")?;
-
-        Ok(response_bytes)
-    }
-
     /// The main loop for sending outbound messages.
     async fn outbound_message_loop<N: ConnectedNetwork<PubKey>>(
         mut receiver: Receiver<OutboundMessage>,
         network: Arc<N>,
+        public_key: PubKey,
+        hotshot_upgrade_lock: UpgradeLock<SeqTypes, V>,
     ) {
         while let Some(message) = receiver.recv().await {
             // Match the message type
             match message {
                 OutboundMessage::Direct(message, recipient) => {
-                    // Send the message directly to the recipient
-                    if let Err(err) = network.direct_message(message, recipient).await {
+                    // Wrap it in the real message type
+                    let message_inner = Message {
+                        sender: public_key,
+                        kind: message,
+                    };
+
+                    // Serialize it
+                    let Ok(message_bytes) = hotshot_upgrade_lock.serialize(&message_inner).await
+                    else {
+                        tracing::error!("Failed to serialize direct message");
```

It looks like we are swallowing the error here but logging an error to indicate there's something really wrong. I think if this is always benign we probably shouldn't log an error and if it isn't we should log the error.

---

I just mean logging the error message, not returning it.

---

Ah okay, yeah in that case I will log it. I also think that it should probably be a warning